workers/java/io/temporal/omes/KitchenSinkWorkflowImpl.java

```diff
@@ -0,0 +1,457 @@
+              .build();
+      Workflow.continueAsNew(options, continueAsNew.getArgumentsList());
+    } else if (action.hasTimer()) {
+      KitchenSink.TimerAction timer = action.getTimer();
+      CancellationScope scope =
+          Workflow.newCancellationScope(
+              () -> {
+                Promise<Void> timerPromise =
+                    Workflow.newTimer(Duration.ofMillis(timer.getMilliseconds()));
+                handlePromise(timerPromise, timer.getAwaitableChoice());
+              });
+      scope.run();
+    } else if (action.hasExecActivity()) {
+      KitchenSink.ExecuteActivityAction activity = action.getExecActivity();
+      launchActivity(activity);
+    } else if (action.hasExecChildWorkflow()) {
+      KitchenSink.ExecuteChildWorkflowAction childWorkflow = action.getExecChildWorkflow();
+      launchChildWorkflow(childWorkflow);
+    } else if (action.hasSetWorkflowState()) {
+      KitchenSink.WorkflowState workflowState = action.getSetWorkflowState();
+      state.putAll(workflowState.getKvsMap());
+    } else if (action.hasAwaitWorkflowState()) {
+      KitchenSink.AwaitWorkflowState awaitWorkflowState = action.getAwaitWorkflowState();
+      Workflow.await(
+          () -> awaitWorkflowState.getValue().equals(state.get(awaitWorkflowState.getKey())));
+    } else if (action.hasNestedActionSet()) {
+      KitchenSink.ActionSet nestedActionSet = action.getNestedActionSet();
+      return handleActionSet(nestedActionSet);
+    } else if (action.hasSendSignal()) {
+      KitchenSink.SendSignalAction sendSignal = action.getSendSignal();
+      ExternalWorkflowStub stub =
+          Workflow.newUntypedExternalWorkflowStub(
+              WorkflowExecution.newBuilder()
+                  .setWorkflowId(sendSignal.getWorkflowId())
+                  .setRunId(sendSignal.getRunId())
+                  .build());
+      CancellationScope scope =
+          Workflow.newCancellationScope(
+              () -> {
+                Promise promise =
+                    Async.procedure(
+                        stub::signal, sendSignal.getSignalName(), sendSignal.getArgsList());
+                handlePromise(promise, sendSignal.getAwaitableChoice());
+              });
+      scope.run();
+    } else if (action.hasUpsertSearchAttributes()) {
+      KitchenSink.UpsertSearchAttributesAction upsertSearchAttributes =
+          action.getUpsertSearchAttributes();
+      Workflow.upsertSearchAttributes(upsertSearchAttributes.getSearchAttributes());
+    } else if (action.hasSetPatchMarker()) {
+      KitchenSink.SetPatchMarkerAction patchMarker = action.getSetPatchMarker();
+      if (Workflow.getVersion(patchMarker.getPatchId(), -1, 1) == 1) {
+        handleAction(patchMarker.getInnerAction());
+      }
+    } else if (action.hasUpsertMemo()) {
+      // TODO(https://github.com/temporalio/sdk-java/issues/623) Java does not support upsert memo.
+      log.warn("Java does not support upsert memo");
```

Comment:
I wonder whether this warning is going to be more annoying than helpful? We're intentionally submitting upsert-memo actions to all langs, so the statement that sdk-java doesn't support them isn't really a warning that something is wrong with the omes run.

Reply:
Yeah. have no issue lowering the log level